query.go

```diff
@@ -0,0 +1,103 @@
+package stream_chat
+
+import (
+	"net/http"
+	"net/url"
+
+	"github.com/getstream/easyjson"
+)
+
+type QueryOption struct {
+	Query map[string]interface{} `json:"-,extra"` // https://getstream.io/chat/docs/#query_syntax
+
+	PaginationOption
+}
+
+type PaginationOption struct {
+	Limit  int `json:"limit,omitempty"`
+	Offset int `json:"offset,omitempty"`
+}
+
+type SortOption struct {
+	Field     string `json:"field"`
+	Direction int    `json:"direction"` // [-1, 1]
+}
+
+type queryUsersRequest struct {
+	FilterConditions *QueryOption  `json:"filter_conditions,omitempty"`
+	Sort             []*SortOption `json:"sort,omitempty"`
+}
+
+type queryUsersResponse struct {
+	Users []*User `json:"users"`
+}
+
+func (c *Client) QueryUsers(q *QueryOption, sort ...*SortOption) ([]*User, error) {
+	qp := queryUsersRequest{
+		FilterConditions: q,
+		Sort:             sort,
+	}
+
+	data, err := easyjson.Marshal(&qp)
+	if err != nil {
+		return nil, err
+	}
+
+	values := make(url.Values)
+	values.Set("payload", string(data))
+
+	var resp queryUsersResponse
+	err = c.makeRequest(http.MethodGet, "users", values, nil, &resp)
+
+	return resp.Users, err
+}
+
+type queryChannelRequest struct {
+	Watch    bool `json:"watch"`
+	State    bool `json:"state"`
+	Presence bool `json:"presence"`
+
+	FilterConditions *QueryOption  `json:"filter_conditions,omitempty"`
+	Sort             []*SortOption `json:"sort,omitempty"`
+}
+
+type queryChannelResponse struct {
+	Channels []queryChannelResponseData `json:"channels"`
+}
+
+type queryChannelResponseData struct {
+	Channel  *Channel         `json:"channel"`
+	Messages []*Message       `json:"messages"`
+	Read     []*ChannelRead   `json:"read"`
+	Members  []*ChannelMember `json:"members"`
+}
+
+func (c *Client) QueryChannels(q *QueryOption, sort ...*SortOption) ([]*Channel, error) {
```

Comment:
Document how the sort order is applied. You could give the return a name (as part of the documentation) like `(hits []*Channel, err error)`.